src/lib/prov/commoncrypto/commoncrypto_hash.cpp

```diff
@@ -71,8 +71,14 @@ class CommonCrypto_HashFunction final : public HashFunction
    private:
       void add_data(const uint8_t input[], size_t length) override
          {
-         if(m_info.update(&m_ctx, input, length) != 1)
-            throw CommonCrypto_Error("CC_" + m_info.name + "_Update");
+         /* update len parameter is 32 bit unsigned integer, feed input in parts */
+         while (length > 0)
+            {
+            CC_LONG update_len = (length > 0xFFFFFFFFUL) ? 0xFFFFFFFFUL : (CC_LONG) length;
```

Comment:
Please use `static_cast` here rather than a C-style cast

Reply:
Just added another commit (05ec523) to use static_cast.